main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -3123,32 +3192,36 @@ object Weeder2 {
       }
     }
 
-    def pickJavaName(tree: Tree): Validation[Name.JavaName, CompilationMessage] = {
+    def pickJavaName(tree: Tree)(implicit sctx: SharedContext): Validation[Name.JavaName, CompilationMessage] = {
       val idents = pickQNameIdents(tree)
       mapN(idents) {
         idents => Name.JavaName(idents, tree.loc)
       }
     }
   }
 
-  private def pickQName(tree: Tree)(implicit sctx: SharedContext): Validation[Name.QName, CompilationMessage] = {
-    mapN(pick(TreeKind.QName, tree))(visitQName)
+  private def pickQName(tree: Tree)(implicit sctx: SharedContext): Option[Name.QName] = {
```

Comment:
this is a key change since it allows `pickQName` to "fail" after `pick` has failed. However, changing this function requires changing all call sites. This is the reason why the PR is so big. I have tried to the best of my ability to carefully consider each call site